### PR TITLE
fix: Change the component reducer for the GSF (to fix navigation issues)

### PR DESCRIPTION
**Pull request by benjaminhuth**
This changes the used component reducer in the examples framework. With the next major release, the default reducer should also be changed in Core.
Also fixes a bug, as the `MaxMomentumReducerLoop` was in fact a `MinMomentumReducerLoop`.

- [x] Implement the change
- [x] Add unit tests

### CODE REVIEW

File: Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp [hunk 1]
```diff
@@ -140,74 +144,96 @@ struct WeightedComponentReducerLoop {
   }
 };
 
-struct MaxMomentumReducerLoop {
```
Comment (andiwand) [inline Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp:143]: this is then typedef'd to `MaxMomentumReducerLoop` right? is this breaking?

Comment (benjaminhuth) [reply Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp:143]: Na I think exchanging an actual class to a typedef that has the same semantics is not breaking, right?

**Review by andiwand — APPROVED** [review]
LGTM

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [8246a1b9f00c10afc33af35e959facf2f206cfca]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42617673) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7978171)!

**Pull request merged**